velox/functions/prestosql/tests/UuidFunctionsTest.cpp

```diff
@@ -108,5 +108,104 @@ TEST_F(UuidFunctionsTest, unsupportedCast) {
       evaluate("cast(123 as uuid())", input), "Cannot cast BIGINT to UUID.");
 }
 
+TEST_F(UuidFunctionsTest, comparisons) {
+  const auto uuidEval = [&](const std::optional<std::string>& lhs,
+                            const std::string& operation,
+                            const std::optional<std::string>& rhs) {
+    return evaluateOnce<bool>(
+        fmt::format("cast(c0 as uuid) {} cast(c1 as uuid)", operation),
+        lhs,
+        rhs);
+  };
+
+  ASSERT_EQ(
+      true,
+      uuidEval(
+          "33355449-2c7d-43d7-967a-f53cd23215ad",
+          "<",
+          "ffffffff-ffff-ffff-ffff-ffffffffffff"));
+  ASSERT_EQ(
```

Comment:
It would be great to have more tests where the lower characters are different.

Could we use some pattern to construct 2 UUIDs such that first 2 bytes of 1st < first 2 bytes of second, 2nd 2 bytes of 1st > 2nd 2 bytes of 2nd, 3rd 2 bytes of 1st < 3rd 2 bytes of 2nd something like :

12342345-3456-4567-5678-678978908901
23451234-4567-3456-6789-567889017890

and then do comparisons by making the first set of bytes equal as you go
So the first comparison would be 
12342345-3456-4567-5678-678978908901 < 23451234-4567-3456-6789-567889017890, 

but the 2nd case would be
00002345-3456-4567-5678-678978908901 > 00001234-4567-3456-6789-567889017890,

 and the 3rd would be 00000000-3456-4567-5678-678978908901< 00000000-4567-3456-6789-567889017890

Reply:
testing against FFFF.... is necessary to make sure it handles the integer overflow property. I do do test a few targeted values where lower portions are different but I can also setup a loop to test what you're describing if you think there should be more.

Reply:
ok added, lmkwyt

Reply:
Thanks !